web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php

```diff
@@ -500,4 +511,39 @@ public static function externalApiBaseUrls(): array {
     return $urls;
   }
 
+  /**
+   * Render the Reader React app.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The HTTP request containing query parameters.
+   *
+   * @return mixed[]
+   *   Render array with the Reader app block.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function reader(Request $request): array {
+
+    $identifier = $request->query->get('identifier');
+    $orderid = $request->query->get('orderid');
+
+    if (!$identifier && !$orderid) {
+      throw new \InvalidArgumentException('Either identifier or orderid must be provided.');
```

Comment:
Please consider changing this to a HTTP exception signaling there is something wrong with the request e.g. `BadRequestHttpException`

A normal exception will result in an error code 500 indicating an internal server error 

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses